microprofile/messaging/core/src/main/java/io/helidon/microprofile/messaging/ProcessorMethod.java

```diff
@@ -54,34 +65,156 @@ public void init(BeanManager beanManager, Config config) {
             switch (getType()) {
                 case PROCESSOR_PUBLISHER_MSG_2_MSG:
                     processor = ReactiveStreams.builder()
-                            .flatMap(in -> ReactiveStreams.<Message<?>>fromPublisher(invoke(in))
-                                    .map(out -> postProcess(in, out)))
+                            .flatMap(in -> {
+                                Message<?> inMsg = (Message<?>) in;
+                                AckCtx ackCtx = AckCtx.create(this, inMsg);
+                                try {
+                                    ackCtx.preAck();
+                                    return ReactiveStreams.fromPublisher(invoke(inMsg))
+                                            .onError(ackCtx::postNack)
+                                            .onErrorResumeWith(this::resumeOnError)
+                                            .peek(out -> ackCtx.postAck())
+                                            .map(MessageUtils::wrap);
+                                } catch (Throwable t) {
```

Comment:
the try/catch seems to be the same all the time. Could we extract at least that?

Reply:
You are right, refactored to standalone method